src/uu/mkdir/src/mkdir.rs

```diff
@@ -216,14 +221,35 @@ fn create_dir(path: &Path, recursive: bool, verbose: bool, is_parent: bool) -> U
                     path.quote()
                 );
             }
-            #[cfg(not(windows))]
-            if is_parent {
-                // directories created by -p have permission bits set to '=rwx,u+wx',
-                // which is umask modified by 'u+wx'
-                chmod(path, (!mode::get_umask() & 0o0777) | 0o0300)?;
+
+            // TODO: Make this macos and freebsd compatible by creating a function to get permission bits from
+            // acl in extended attributes
+            #[cfg(not(any(target_os = "windows", target_os = "macos", target_os = "freebsd")))]
```

Comment:
what about ?

it makes it clear that the call is always chmod and we just have different mode for each platform?


 #[cfg(not(any(target_os = "windows", target_os = "macos", target_os = "freebsd")))]
Comment
let new_mode = if is_parent {
    (!mode::get_umask() & 0o777) | 0o300
} else {
    mode
};

#[cfg(not(any(target_os = "windows", target_os = "macos", target_os = "freebsd")))]
let new_mode = if !path_exists {
    let acl_perm_bits = uucore::fsxattr::get_acl_perm_bits_from_xattr(path);
    new_mode | acl_perm_bits
} else {
    new_mode
};

#[cfg(windows)]
let new_mode = mode;

chmod(path, new_mode)?;

Ok(())

Reply:
Yeah that feels more deliberate. 

We replace the large os exceptions with this 

```
 #[cfg(all(unix,not(target_os="linux")))]
Comment
let new_mode = if is_parent {
    (!mode::get_umask() & 0o777) | 0o300
} else {
    mode
};

 #[cfg(all(unix,target_os="linux"))]
let new_mode = if !path_exists {
    let acl_perm_bits = uucore::fsxattr::get_acl_perm_bits_from_xattr(path);
    new_mode | acl_perm_bits
} else {
    new_mode
};

#[cfg(windows)]
let new_mode = mode;

chmod(path, new_mode)?;
```

From the earlier CI checks , i don't think android xattr works either, it does seem its because the tag values i am using is different for android so the test kept failing. There seems to be some differences in every platform, which would require reading into their source code. Ideally we could have used the "exacl" crate but it wouldn't work on most systems as most distros ship without libacl.